postgres_test.go

```diff
@@ -0,0 +1,72 @@
+package ipam
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestDatasource(t *testing.T) {
+	tests := []struct {
+		name     string
+		host     string
+		port     string
+		user     string
+		password string
+		dbname   string
+		sslmode  SSLMode
+		want     string
+		wantErr  bool
+	}{
+		{
```

Comment:
you missed your special use case with an empty password in the tests here

Reply:
Thanks. I've amended the second commit and also used subtests to consume the `name` field from the test cases.

Reply:
Tests failed

Reply:
I cannot reproduce the same test failure (tcp read).

On my Thinkpad, the tests pass via `make test`. 

On my desktop, the tests time out, even on the `master` branch, even after extending the timeout to 20m.

Do you have any tips for reproducing this test environment or providing a way for me to re-run the Docker Build Action here on Github?

https://gist.github.com/ralfonso/d00c6a32feaae3dae83ca36f85152a53

Reply:
Running `docker system prune` seems to have resolved the timeouts on my desktop. The only place this is failing is in GH Actions.